src/app/error.tsx

```diff
@@ -1,9 +1,29 @@
-'use client' // Error components must be Client Components
+'use client'
 
-export default function GlobalError({ error }: { error: Error & { digest?: string } }) {
+import React from 'react'
+
+const getErrorMessage = (error: Error & { digest: string }) => {
+  if (error.message) {
+    return error.message
+  }
+  return 'Something went wrong'
```

Comment:
```suggestion
const getErrorMessage = (error: Error & { digest: string }) =>
  error.message ?? 'Something went wrong'
```

Reply:
Can we also make sure this won't expose sensitive / technical information to this component?

This block catches all errors and displays their error message. All `Error` objects have a message so `if (message) -> "XYZ" || "Something went wrong"` really doesn't make sense. A non-technical user seeing something like "Fetch failed" or a long cryptic error message for a state update error would not be preferable - instead we would like to show "Something went wrong" in those cases.